src/Umbraco.Web.Website/Routing/PublicAccessRequestHandler.cs

```diff
@@ -65,6 +67,19 @@ public async Task<UmbracoRouteValues> RewriteForPublishedContentAccessAsync(Http
                 {
                     _logger.LogDebug("EnsurePublishedContentAccess: Page is protected, check for access");
 
+                    // manually authenticate the request
+                    AuthenticateResult authResult = await httpContext.AuthenticateAsync(IdentityConstants.ApplicationScheme);
+                    if (authResult.Succeeded)
+                    {
+                        // set the user to the auth result. we need to do this here because this occurs
+                        // before the authentication middleware.
+                        // NOTE: It would be possible to just pass the authResult to the HasMemberAccessToContentAsync method
+                        // instead of relying directly on the user assigned to the http context, and then the auth middleware
+                        // will run anyways and assign the user. Perhaps that is a little cleaner, but would require more code
+                        // changes right now, and really it's not any different in the end result.
+                        httpContext.User = authResult.Principal;
```

Comment:
Just to align with the codebase.
```suggestion
                        httpContext.SetPrincipalForRequest(authResult.Principal);
```